Sources/mParticle-Appboy/MPKitAppboy.m

```diff
@@ -432,19 +433,18 @@ - (MPKitExecStatus *)routeCommerceEvent:(MPCommerceEvent *)commerceEvent {
         }
         
         if (_configuration[bundleProductsWithCommerceEvents] && [_configuration[bundleProductsWithCommerceEvents] boolValue]) {
-            NSMutableArray *productArray = [[NSMutableArray alloc] init];
-            for (MPProduct *product in products) {
-                // Add attributes from the products themselves
-                NSMutableDictionary *productDictionary = [[product beautifiedDictionaryRepresentation] mutableCopy];
-                                
-                // Adds the product dictionary to the product array being supplied to Braze
-                if (productDictionary) {
-                    [productArray addObject:productDictionary];
-                }
-            }
+            NSArray *productArray = [self getProductListParameters:products];
             if (productArray.count > 0) {
                 [properties setValue:productArray forKey:productKey];
             }
+            NSArray *promotionArray = [self getPromotionListParameters:commerceEvent.promotionContainer.promotions];
+            if (promotionArray.count > 0) {
+                [properties setValue:promotionArray forKey:promotionKey];
+            }
+            NSArray *impressionArray = [self getImpressionListParameters:commerceEvent.impressions];
+            if (impressionArray.count > 0) {
+                [properties setValue:impressionArray forKey:impressionKey];
+            }
```

Comment:
If I'm reading this correctly, setting up the promotions and impressions here fall under the block for if there is a purchase action.  Is this necessary?  If it is a purchase action, it cannot contain promotions/impressions. It's only if it's a nonpurchase commerce event that it could potentially contain promotions/impressions.

Reply:
It's possible for clients to set promotions/impressions on purchase events in iOS but highly unlikely. I'm fine removing the possibility here if that's what you'd prefer.